### PR TITLE
Switched to PSR-15 middleware for Neos 7.0

### PR DESCRIPTION
**Pull request by MarcoPNS**
Just updated my fork to Neos 7.0 (the one with the resolutionHost feature #3) - It may be helpful to apply the changes here as well.

### CODE REVIEW

File: Classes/Http/DetectContentSubgraphComponent.php [hunk 1]
```diff
@@ -15,15 +15,21 @@
...
-use Neos\Flow\Mvc\Routing\RoutingComponent;
+use Neos\Flow\Http\ServerRequestAttributes;
+
 use Flowpack\Neos\DimensionResolver\Http\ContentDimensionDetection\DimensionPresetDetectorResolver;
 
 /**
  * The HTTP component for detecting the requested dimension space point
  */
-final class DetectContentSubgraphComponent implements Http\Component\ComponentInterface
+final class DetectContentSubgraphComponent implements MiddlewareInterface
```
Comment (bwaidelich) [inline Classes/Http/DetectContentSubgraphComponent.php]: This should probably be renamed to `DetectContentSubgraphMiddleware` to avoid confusion!?

**Review by bwaidelich — COMMENTED** [review]
I can't test this right now, but thanks a lot for putting so much effort into this It looks great by reading!

I just left a nitpick comment for now

**Comment by MarcoPNS** [comment]
@bwaidelich Thank you for your feedback! Just renamed it.

**Review by bwaidelich — COMMENTED** [review]
did you rename the files, too?  since it doesn't seem so but I might got confused by the github ui

**Comment by MarcoPNS** [comment]
The file has also been renamed. What still needs to be rewritten are the tests. I have not yet updated them.

File: Tests/Functional/Http/DetectContentSubgraphComponentTest.php [hunk 2]
```diff
@@ -16,12 +16,12 @@
...
 use Neos\Flow\Tests\FunctionalTestCase;
 use Flowpack\Neos\DimensionResolver\Http\ContentDimensionResolutionMode;
-use Flowpack\Neos\DimensionResolver\Http\DetectContentSubgraphComponent;
+use Flowpack\Neos\DimensionResolver\Http\DetectContentSubgraphMiddleware;
 
 /**
  * Test case for the BackendUriDimensionPresetDetector
  */
-class DetectContentSubgraphComponentTest extends FunctionalTestCase
+class DetectContentSubgraphMiddlewareTest extends FunctionalTestCase
```
Comment (bwaidelich) [inline Tests/Functional/Http/DetectContentSubgraphComponentTest.php:24]: File needs to be renamed accordingly

File: Tests/Unit/Http/ContentDimensionLinking/DimensionPresetLinkProcessorResolverTest.php [hunk 3]
```diff
@@ -17,7 +17,7 @@
 use Flowpack\Neos\DimensionResolver\Tests\Unit\Http\ContentDimensionLinking\Fixtures\ValidDummyDimensionPresetLinkProcessor;
 
 /**
- * Test case for the DetectContentSubgraphComponent
+ * Test case for the DetectContentSubgraphMiddleware
```
Comment (bwaidelich) [inline Tests/Unit/Http/ContentDimensionLinking/DimensionPresetLinkProcessorResolverTest.php:20]:
```suggestion
 * Test case for the DimensionPresetLinkProcessorResolver
```

**Comment by bwaidelich** [comment]
> The file has also been renamed

ah right. It's just the test file that hasn't been renamed it seems

**Comment by putzwasser** [comment]
I updated to 7.0 and it broke neos-dimensionresolver. Found this PR. Is this PR ready for merging? :)

**Comment by MarcoPNS** [comment]
just checked if everything is renamed and it should be save to merge.

**Comment by gitvitox** [comment]
Any updates on this? @bwaidelich

**Review by bwaidelich — APPROVED** [review]
no time to test atm, but this looks good by reading. Thanks!

**Review by albe — APPROVED** [review]
Looks okay as far as I can tell.

**Comment by albe** [comment]
Let's merge 🤞

**Comment by putzwasser** [comment]
Could you please tag this commit, so composer will install it without whining about minimum stability requirements :D ?

**Pull request merged**